libraries/stdlib/samples/test/samples/collections/collections.kt

```diff
@@ -337,6 +337,94 @@ class Collections {
 
     class Transformations {
 
+        @Sample
+        fun associate() {
+            data class Person(val firstName: String, val lastName: String)
+
+            val scientists = listOf(Person("Grace", "Hopper"), Person("Jacob", "Bernoulli"), Person("Johann", "Bernoulli"))
+
+            val byLastName = scientists.associate { it.lastName to it.firstName }
```

Comment:
While this reads good, it can be replaced with more efficient, though little less pretty  `associate({ it.lastName }, { it.firstName })`. The real strength of _this_ overload is that it can map the item somehow and then use the properties of that result as key and value.
For example, suppose we have a list of names as `listOf("Grace Hopper", "Jacob Bernoulli", ...)`. Then we can build the same map as in this sample as following:
```
names.associate { it.split(" ").let { (firstName, lastName) -> lastName to firstName } }
```
Without this overload we would have to split the name string twice or use an intermediate `map` operation.

Reply:
Alright, I'll use that String example.